lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -307,8 +306,7 @@ def compile_mixins(constant)
               Module.ancestors
             end
 
-          interesting_ancestors =
-            constant.ancestors.reject { |mod| ignorable_ancestors.include?(mod) }
+          interesting_ancestors = constant.ancestors - ignorable_ancestors
```

Comment:
Isn't `Array#-` also using `hash`?

From the docs:

```
Array Difference

Returns a new array that is a copy of the original array, removing any items that also appear in other_ary. The order is preserved from the original array.

It compares elements using their hash and eql? methods for efficiency.
```

Reply:
It turns out that 2.4 uses `hash` but anything above uses `eql?` I guess. I need to find a different solution if we want to keep supporting 2.4.

Reply:
Ok, I changed the implementation. Can you review again please?